### PR TITLE
Add voting machine

### PR DESCRIPTION
**Pull request by spaugh**

### CODE REVIEW

**Comment by spaugh** [comment]
@intoverflow i changed the layout quite a bit. and, tbh, i didn't dig in too deeply. let me know if you see anything insane.

File: voting-machine/host/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+env_logger = "0.9.0"
+log = "0.4.17"
+clap = "3.1.18"
+ctor = "0.1"
+tempfile = "3.3"
+voting-machine-methods = { path = "../methods" }
+voting-machine-core = { path = "../core" }
+risc0-zkvm-core = "0.7"
+risc0-zkvm-host = "0.7"
+risc0-zkvm-serde = "0.7"
```
Comment (flaub) [inline voting-machine/host/Cargo.toml:16]: Please keep these sorted

File: voting-machine/core/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+risc0-zkvm-core = { version = "0.7", default-features = false, features = ["pure"] }
+serde = { version = "1.0", default-features = false }
+
+[dev-dependencies]
+voting-machine-methods = { path = "../methods" }
+risc0-zkvm-serde = "0.7"
```
Comment (flaub) [inline voting-machine/core/Cargo.toml:12]: `[dev-dependencies]` are meant to be used for tests, examples, and benchmarks. I didn't see any of those in this crate.

File: voting-machine/host/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+use risc0_zkvm_host::{Prover, Receipt, Result};
+use risc0_zkvm_serde::{from_slice, to_vec};
+use std::fs;
+use tempfile::{NamedTempFile, TempPath};
+use voting_machine_core::{Ballot, SubmitBallotCommit, SubmitBallotParams, SubmitBallotResult};
+use voting_machine_core::{
+    FreezeVotingMachineCommit, FreezeVotingMachineParams, FreezeVotingMachineResult,
+};
+use voting_machine_core::{InitializeVotingMachineCommit, VotingMachineState};
+use voting_machine_methods::{FREEZE_ID, FREEZE_PATH, INIT_ID, INIT_PATH, SUBMIT_ID, SUBMIT_PATH};
```
Comment (flaub) [inline voting-machine/host/src/lib.rs:24]: Two kinds of groupings for uses:

* by source
  * std, alloc, core
  * third party
  * current crate uses
* group together common prefixes

```suggestion
use std::fs;

use risc0_zkvm_host::{Prover, Receipt, Result};
use risc0_zkvm_serde::{from_slice, to_vec};
use tempfile::{NamedTempFile, TempPath};
use voting_machine_core::{Ballot, SubmitBallotCommit, SubmitBallotParams, SubmitBallotResult, FreezeVotingMachineCommit, FreezeVotingMachineParams, FreezeVotingMachineResult, InitializeVotingMachineCommit, VotingMachineState};
use voting_machine_methods::{FREEZE_ID, FREEZE_PATH, INIT_ID, INIT_PATH, SUBMIT_ID, SUBMIT_PATH};
```

File: voting-machine/host/Cargo.toml (hunk 1 above)
Comment (spaugh) [reply voting-machine/host/Cargo.toml:16]: In what manner? I just googled and found two different strategies - sorted alphabetically and sorted alphabetically within "class" of dependency.

File: voting-machine/core/Cargo.toml (hunk 2 above)
Comment (spaugh) [reply voting-machine/core/Cargo.toml:12]: Hmm yup looks like a bad copy/paste job. Good catch.

**Review by intoverflow — APPROVED** [review]
🤘

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -25,6 +25,8 @@ jobs:
       - uses: actions-rs/toolchain@v1
         with:
           toolchain: stable
+      - run: cargo fmt --check
```
[the remark this answers is not in the document]
Comment (flaub) [reply .github/workflows/main.yml:28]: awesome thanks, good idea to add this to all the CIs

File: .rustfmt.toml [hunk 5]
```diff
@@ -0,0 +1,2 @@
+group_imports = "StdExternalCrate"
+imports_granularity = "Module"
```
Comment (flaub) [inline .rustfmt.toml:2]: ❤️

**Pull request merged**